### PR TITLE
Enumerate the OCI_ENV_PATH automatically.

### PR DESCRIPTION
**Pull request by jctanner**
Forgetting to set OCI_ENV_PATH or not setting it correctly is a pain point with using oci-env.

### CODE REVIEW

File: client/oci_env/utils.py [hunk 1]
```diff
@@ -1,19 +1,45 @@
...
 from urllib import request
+from logzero import logger
 
 
 def get_oci_env_path():
-    return os.environ.get("OCI_ENV_PATH", os.getcwd())
+    """This returns the root directory of the oci-env checkout."""
+    if os.environ.get("OCI_ENV_PATH"):
+        logger.info('USING OCI_ENV_PATH FROM ENVIRONMENT: {os.environ.get("OCI_ENV_PATH")}')
+        return os.environ.get("OCI_ENV_PATH")
```
Comment (decko) [inline client/oci_env/utils.py:15]: What about:
```suggestion
    if OCI_ENV_PATH := os.environ.get("OCI_ENV_PATH"):
        logger.info('USING OCI_ENV_PATH FROM ENVIRONMENT: {OCI_ENV_PATH}')
        return OCI_ENV_PATH
```

File: client/oci_env/utils.py [hunk 2]
```diff
@@ -1,19 +1,45 @@
-from genericpath import isfile
 import os
-import subprocess
 import pathlib
+import subprocess
 import time
 
+from genericpath import isfile
 from urllib import request
+from logzero import logger
```
Comment (gerrod3) [inline client/oci_env/utils.py:8]: Do we really need this logging library?

File: .github/workflows/ci.yaml [hunk 3]
```diff
@@ -62,9 +65,9 @@ jobs:
             sudo dpkg -i containernetworking-plugins_1.1.1+ds1-1_amd64.deb
           fi
 
-      - name: (Ubuntu) Install docker compose
-        if: matrix.TEST == 'docker' && matrix.os == 'ubuntu-latest'
-        uses: docker-practice/actions-setup-docker@1.0.11
+      #- name: (Ubuntu) Install docker compose
+      #  if: matrix.TEST == 'docker' && matrix.os == 'ubuntu-latest'
+      #  uses: docker-practice/actions-setup-docker@1.0.11
```
Comment (gerrod3) [inline .github/workflows/ci.yaml:70]: Why are these commented out?

Comment (jctanner) [reply .github/workflows/ci.yaml:70]: https://github.com/ansible/galaxy_ng/pull/1800

Comment (decko) [reply .github/workflows/ci.yaml]: although it says `Install docker compose`, it installs docker and docker-compose. I believe you need this to test oci-env using docker and yet you can install the new `docker-compose` binary.

Comment (jctanner) [reply .github/workflows/ci.yaml]: docker is already installed in the ubuntu vms provided by github actions. Also... the tests are passing without this.

File: client/oci_env/utils.py [hunk 4]
```diff
@@ -1,19 +1,50 @@
...
 import os
-import subprocess
 import pathlib
+import subprocess
 import time
 
+from genericpath import isfile
 from urllib import request
 
+#from logzero import logger
```
Comment (gerrod3) [inline client/oci_env/utils.py:9]:
```suggestion
```

File: client/oci_env/utils.py [hunk 5]
```diff
@@ -1,19 +1,50 @@
...
+from oci_env.logger import logger
+
 
 def get_oci_env_path():
-    return os.environ.get("OCI_ENV_PATH", os.getcwd())
+    """This returns the root directory of the oci-env checkout."""
+
+    print(type(logger))
```
Comment (gerrod3) [inline client/oci_env/utils.py:16]:
```suggestion
```

File: client/oci_env/utils.py [hunk 6]
```diff
@@ -1,19 +1,50 @@
...
+        stderr=subprocess.PIPE
+    )
+    if pid.returncode != 0:
+        cwd = os.getcwd()
+        logger.warning(f'USING CWD {cwd} FOR OCI_ENV_PATH BECAUSE OF GIT CMD FAILURE {pid.stdout}')
+        return cwd
+
+    gitroot = pid.stdout.decode('utf-8').strip()
+    logger.info(f'USING {gitroot} FOR OCI_ENV_PATH BASED ON GIT CMD OUTPUT')
+    return gitroot.rstrip('/') + '/'
```
Comment (gerrod3) [inline client/oci_env/utils.py:40]: Is the path suppose to have a trailing slash?

File: client/oci_env/logger.py [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+logging.setLogRecordFactory(LogColor)
+logger = logging.getLogger('oci_env')
+logger.setLevel(logging.INFO)
+console_handler = logging.StreamHandler()
+console_handler.setFormatter(formatter)
+
+
+if os.environ.get("OCI_ENV_DEBUG"):
+	logger.setLevel(logging.DEBUG)
+	console_handler.setLevel(logging.DEBUG)
```
Comment (gerrod3) [inline client/oci_env/logger.py:42]: Is this double indented?

File: .github/workflows/ci.yaml [hunk 8]
```diff
@@ -30,11 +30,14 @@ jobs:
         with:
           python-version: "3.8"
 
+      - name: DEBUG VAR
+        run: echo "${GITHUB_WORKSPACE}"
+
```
Comment (gerrod3) [inline .github/workflows/ci.yaml:35]: Is this still needed?

**Review by gerrod3 — CHANGES_REQUESTED** [review]

File: client/oci_env/utils.py (hunk 6 above)
Comment (newswangerd) [reply client/oci_env/utils.py:37]: I would say no trailing slash. My path is set to `/Users/dnewswan/code/hub/oci_env` and there are a lot of places where the path is referenced like this: https://github.com/pulp/oci_env/blob/main/base/compose.yaml#L17

I'm not totally sure if that would work with a trailing /

File: client/oci_env/logger.py [hunk 9]
```diff
@@ -0,0 +1,46 @@
+# logger.py - reimplement some functionality of logzero
+
+import logging
```
Comment (newswangerd) [inline client/oci_env/logger.py:3]: @jctanner ++ for better logging!

File: client/oci_env/utils.py [hunk 10]
```diff
@@ -1,19 +1,47 @@
...
+from oci_env.logger import logger
+
 
 def get_oci_env_path():
-    return os.environ.get("OCI_ENV_PATH", os.getcwd())
+    """This returns the root directory of the oci-env checkout."""
+
+    if OCI_ENV_PATH := os.environ.get("OCI_ENV_PATH"):
+        logger.info('USING OCI_ENV_PATH FROM ENV: {OCI_ENV_PATH}')
+        return OCI_ENV_PATH
```
Comment (newswangerd) [inline client/oci_env/utils.py:17]: i think it would be good to strip the trailing slash from this as well

**Review by newswangerd — APPROVED** [review]
this is great! Just clean up the trailing slashes and you're good to go

**Review by decko — APPROVED** [review]

**Review by gerrod3 — APPROVED** [review]

**Pull request merged**